libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -97,321 +97,412 @@ const AP_Param::GroupInfo AC_AttitudeControl::var_info[] = {
     AP_GROUPEND
 };
 
-void AC_AttitudeControl::relax_bf_rate_controller()
+// Set output throttle and disable stabilization
+void AC_AttitudeControl::set_throttle_out_unstabilized(float throttle_in, bool reset_attitude_control, float filter_cutoff)
 {
+    _throttle_in = throttle_in;
+    _motors.set_throttle_filter_cutoff(filter_cutoff);
+    if (reset_attitude_control) {
+        relax_attitude_controllers();
+    }
+    _motors.set_throttle(throttle_in);
+    _angle_boost = 0.0f;
+}
+
+// Ensure attitude controller have zero errors to relax rate controller output
+void AC_AttitudeControl::relax_attitude_controllers()
+{
+    // TODO add _ahrs.get_quaternion()
+    _attitude_target_quat.from_rotation_matrix(_ahrs.get_rotation_body_to_ned());
+    _attitude_target_quat.to_euler(_attitude_target_euler_angle.x, _attitude_target_euler_angle.y, _attitude_target_euler_angle.z);
+    _attitude_target_ang_vel = _ahrs.get_gyro();
+    _attitude_target_euler_angle = Vector3f(_ahrs.roll, _ahrs.pitch, _ahrs.yaw);
```

Comment:
Why get Euler values from the quaternion if we then overwrite them with this Vector?